daemons/pacemakerd/pcmkd_messages.c

```diff
@@ -39,44 +57,52 @@ pcmk_handle_ping_request(pcmk__client_t *c, xmlNode *msg, uint32_t id)
                    (long long) subdaemon_check_progress);
     crm_xml_add(ping, XML_PING_ATTR_STATUS, "ok");
     reply = create_reply(msg, ping);
+
     free_xml(ping);
-    if (reply) {
-        if (pcmk__ipc_send_xml(c, id, reply, crm_ipc_server_event) !=
-                pcmk_rc_ok) {
-            crm_err("Failed sending ping reply to client %s",
-                    pcmk__client_name(c));
-        }
-        free_xml(reply);
+
+    pcmk__ipc_send_ack(request->ipc_client, request->ipc_id, request->ipc_flags,
+                       "ack", NULL, CRM_EX_INDETERMINATE);
```

Comment:
It doesn't really matter in this case since all we do is create XML before this, but as a general practice, it's a good idea to send the ack as soon as possible to minimize the chance of the client timing out while waiting for it.